File: app/src/main/java/com/erikzuo/portfolioandroidapp/data/remote/GithubService.java
package com.erikzuo.portfolioandroidapp.data.remote;

/**
 * Created by Soprano on 7/02/2018.
 */

import com.erikzuo.portfolioandroidapp.data.model.Repo;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Github REST API access points
 */
public interface GithubService {

    @GET("/users/{username}/repos")
    Observable<List<Repo>> getRepos(@Path("username") String username);
}

File: app/src/main/java/com/erikzuo/portfolioandroidapp/data/remote/GithubService.kt
package com.erikzuo.portfolioandroidapp.data.remote

/**
 * Created by YifanZuo on 7/02/2018.
 */

import com.erikzuo.portfolioandroidapp.data.model.Repo

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Github REST API access points
 */
interface GithubService {

    @GET("/users/{username}/repos")
    fun getRepos(@Path("username") username: String): Observable<List<Repo>>
}
